pkg/middlewares/metrics/recorder.go

```diff
@@ -34,12 +44,27 @@ type responseRecorderWithCloseNotify struct {
 	*responseRecorder
 }
 
+type responseRecorderWithH2push struct {
+	*responseRecorder
+}
+
+type responseRecorderWithCloseNotifyAndH2push struct {
+	*responseRecorder
+	*responseRecorderWithCloseNotify
+	*responseRecorderWithH2push
```

Comment:
I did it this way in order to not duplicate the methods below, however I don't know if this is the best approach.